lib/fileOperations.js

```diff
@@ -196,11 +198,55 @@ function writeFile(path, data, options, callback) {
   }
 }
 
+function mkdirp(dirpath, mode, callback) {
+  if (typeof mode === 'function') {
+    callback = mode;
+    mode = undefined;
+  }
+
+  var m = mode || DEFAULT_DIR_MODE;
+  var cb = callback || function() {};
+  dirpath = path.resolve(dirpath);
+
+  fs.mkdir(dirpath, m, function(er) {
```

Comment:
The initial fs.mkdir call ends the recursion on success.  If a parent directory doesn't exist, it recursively calls 'mkdirp' up the path until fs.mkdir succeeds.  Then calls mkdir back down the path.